public/app/features/trails/MetricSelectScene.tsx

```diff
@@ -241,6 +244,15 @@ export class MetricSelectScene extends SceneObjectBase<MetricSelectSceneState> {
           continue;
         }
         const panel = getPreviewPanelFor(metric.name, index, currentFilterCount);
+
+        const vizPanel = panel.state.body;
+        if (vizPanel instanceof VizPanel) {
+          trail.getMetricMetadata(metric.name).then((metadata) => {
+            const description = getMetricDescription(metadata);
+            vizPanel.setState({ description });
+          });
+        }
```

Comment:
I would make this function async and wait for await call, and pass in the description for getPreviewPanelFor , that way we don't have this double rendering. the metadata call is fast (and should be cached)

Reply:
Fixed